### ISSUE TITLE
network: use `io.LimitReader` whenever reading from the network

### ISSUE THREAD

**Issue by qdm12** [ChainSafe/gossamer#2444]
Use [io.LimitReader](https://pkg.go.dev/io#LimitReader) to limit maliciously large network messages from blowing up our memory.

I spotted the following to change:

- `io.ReadAll(r.Body)` usage in `dot/rpc/subscription/websocket.go`
- `json.NewDecoder(r.Body)` in `dot/rpc/json2/server.go`

**Comment by noot** [comment]
@qdm12 we already read into a limited size buffer only up until a certain size, so would that prevent this? at least for the network

**Comment by qdm12** [comment]
Indeed, my bad. I updated the issue body with only 2 rpc related issues I spotted.

Essentially, even if we read just with a single-byte byte slice from a reader, that reader has bytes already in memory. So we should pay attention to limit the amount of data that reader can have at the network entrypoints, using `io.LimitReader`. Unfortunately, I'm not 100% sure about where all the network external 'entrypoints' are yet.
